app/src/main/java/com/example/myapplication/ui/professorsList/ProfessorListViewModel.kt

```diff
@@ -62,4 +65,14 @@ class ProfessorListViewModel @Inject constructor(
         _errorMessageLiveData.value = throwable.message
         _viewStateLiveData.value = ViewState.ERROR
     }
+
+    fun onProfessorSelected(professor: Professor) {
+        selectProfessorUseCase(professor).also {
```

Comment:
instead of using use cases and passing data to lower layers you can use parentViewModel to set selected professor and then trigger a LiveData which is been observing in ProfessorDetailFragment.